v1/contracts/VotingToken.sol

```diff
@@ -0,0 +1,33 @@
+pragma solidity ^0.5.0;
+
+import "./MultiRole.sol";
+
+import "openzeppelin-solidity/contracts/drafts/ERC20Snapshot.sol";
+
+
+/**
+ * @title UMA voting token
+ * @dev Supports snapshotting and allows the Oracle to mint new tokens as rewards.
+ */
+contract VotingToken is ERC20Snapshot, MultiRole {
+
+    enum Roles {
+        // Can set the minter.
+        Governance,
+        // The Oracle contract (currently named Voting.sol) can mint new tokens as voting rewards.
+        Minter
+    }
+
+    constructor() public {
+        _createExclusiveRole(uint(Roles.Governance), uint(Roles.Governance), msg.sender);
+        _createExclusiveRole(uint(Roles.Minter), uint(Roles.Governance), msg.sender);
```

Comment:
Why does the minter need to be exclusive? I was thinking we might ultimately want the flexibility to mint outside of the inflationary process during the initial token distribution process. WDYT?

Reply:
Seems reasonable.